src/main/java/org/openrewrite/java/template/processor/RefasterTemplateProcessor.java

```diff
@@ -607,48 +606,25 @@ private Set<String> getImportsAsStrings(Map<TemplateDescriptor, Set<String>> imp
                             }
                             String methodName = method.name.toString();
                             methodName = methodName.equals("<init>") ? "<constructor>" : methodName;
-                            usesVisitors.add("new UsesMethod<>(\"" + method.owner.getQualifiedName().toString() + ' ' + methodName + "(..)\", true)");
+                            usesVisitors.add(new Precondition.Rule("new UsesMethod<>(\"" + method.owner.getQualifiedName().toString() + ' ' + methodName + "(..)\", true)"));
                         }
 
-                        preconditions.put(beforeTemplate.method.name.toString() + (arity == 1 ? "" : "$" + i), usesVisitors);
-                    }
-                }
-
-                if (preconditions.size() == 1) {
-                    return joinPreconditions(preconditions.values().iterator().next(), "and", indent + 4);
-                } else if (preconditions.size() > 1) {
-                    Set<String> common = new LinkedHashSet<>();
-                    for (String dep : preconditions.values().iterator().next()) {
-                        if (preconditions.values().stream().allMatch(v -> v.contains(dep))) {
-                            common.add(dep);
+                        if (!usesVisitors.isEmpty()) {
+                            preconditions.put(beforeTemplate.method.name.toString() + (arity == 1 ? "" : "$" + i), usesVisitors);
                         }
```

Comment:
Should this instead be 
```suggestion
                        if (!usesVisitors.isEmpty()) {
                            preconditions.put(beforeTemplate.method.name.toString() + (arity == 1 ? "" : "$" + i), usesVisitors);
                        } else {
                            return null; // At least one of the before templates has no preconditions, so we can not use any preconditions
                        }
```

Reply:
You are right; my test did not caught this problem, because there was alway the `java.io.PrintStream` type... For example:

```java
@BeforeTemplate
void _int(int actual) {
  System.out.println(actual);
}
```        

I will change the tests accordingly!